internal/command/doctor/app_checks.go

```diff
@@ -71,12 +64,20 @@ func NewAppChecker(ctx context.Context, jsonOutput bool, color *iostreams.ColorS
 
 	if !appCompact.Deployed && appCompact.PlatformVersion != "machines" {
 		ac.lprint(color.Yellow, "%s app has not been deployed yet. Skipping app checks. Deploy using `flyctl deploy`.\n", appName)
-		return nil
+		return nil, nil
 	}
 
 	ac.app = appCompact
 	ac.appConfig = appconfig.ConfigFromContext(ctx)
-	return ac
+
+	if ac.appConfig == nil {
+		ac.appConfig, err = appconfig.FromRemoteApp(ctx, ac.app.Name)
```

Comment:
is flapsClient in the context? is common for it to be missing and the command crashes when trying to access the configuration for a non-deployed app.

the easiest way to check is something like:
```
fly apps create --machines my-app-name
fly doctor -a my-app-name
```

Reply:
It doesn't crash:

```
$ fly apps create dangra-t9 -o personal
New app created: dangra-t9

$ fly doctor -a dangra-t9
Testing authentication token... PASSED
Testing flyctl agent... PASSED
Testing local Docker instance... PASSED
Pinging WireGuard gateway (give us a sec)... PASSED
dangra-t9 app has not been deployed yet. Skipping app checks. Deploy using `flyctl deploy`.
```